rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -430,19 +430,19 @@ rosbag2_storage::BagMetadata SqliteStorage::get_metadata()
 
   auto statement = database_->prepare_statement(
     "SELECT name, type, serialization_format, COUNT(messages.id), MIN(messages.timestamp), "
-    "MAX(messages.timestamp) "
+    "MAX(messages.timestamp), offered_qos_profiles "
```

Comment:
why are we needing this change here for the reindexing?

Reply:
This is because the original metadata file includes the QOS profiles of the topics, so I have to query them from the database. I did some searching around, and I don't think this change affects anything else in the project, though if you have an alternate way to query the QOS profiles from the database, that might be preferable.